source/frontend/src/hooks/useComposedProperties.ts

```diff
@@ -52,18 +52,35 @@ export const useComposedProperties = ({
   const executeGetApiProperty = getApiPropertyWrapper.execute;
   const executeGetPropertyWfs = getPropertyWfsWrapper.execute;
   const executeGetPropertyAssociations = getPropertyAssociationsWrapper.execute;
+  const executeGetAllFeatures = getAllFeaturesWrapper.execute;
 
   useEffect(() => {
     if (id !== undefined && !isNaN(id)) {
       typeCheckWrapper(() => executeGetApiProperty(id), PROPERTY_TYPES.PIMS_API);
       typeCheckWrapper(() => executeGetPropertyWfs(id), PROPERTY_TYPES.PIMS_GEOSERVER);
       typeCheckWrapper(() => executeGetPropertyAssociations(id), PROPERTY_TYPES.ASSOCIATIONS);
     }
+    if (!!retrievedPid) {
+      typeCheckWrapper(
+        () =>
+          executeGetAllFeatures({ PID: retrievedPid }, { forceSimplePid: true, timeout: 30000 }),
+        PROPERTY_TYPES.PARCEL_MAP,
+      );
+    }
+    if (!!retrievedPin) {
+      typeCheckWrapper(
+        () => executeGetAllFeatures({ PIN: retrievedPin }, { timeout: 30000 }),
+        PROPERTY_TYPES.PARCEL_MAP,
+      );
+    }
   }, [
     executeGetApiProperty,
     executeGetPropertyAssociations,
     executeGetPropertyWfs,
+    executeGetAllFeatures,
     id,
+    retrievedPid,
```

Comment:
@Maheshbabu28 This is not a good idea unfortunately. What this will result in is:
1) the first three requests will be executed because the id is defined.
2) those requests will complete which will populate the pid/pin.
3) the first three requests will execute again for the same id, then the other two requests will execute.

The useffect for the later two requests should be split out. to avoid this. Actually, why wouldn't this be included in the useeffect below which is already using the retrievedPid/pin?

Reply:
pid is accepting as prop but missed also extracted from api response. You are right its better places in the useeffect below, moved.